core/cmd/local_client_test.go

```diff
@@ -43,10 +44,20 @@ func TestClient_RunNodeShowsEnv(t *testing.T) {
 	eth := app.MockEthClient()
 	eth.Register("eth_getTransactionCount", `0x1`)
 
-	assert.NoError(t, client.RunNode(c))
+	// Start RunNode in a goroutine, it will block until we resume the runner
+	go func() {
+		assert.NoError(t, client.RunNode(c))
```

Comment:
I understand what's going on here, but deep down I don't think it should be needed.

We're taking something that we believe to be synchronous (`client.RunNode`), placing it in a goroutine, and then blocking until it signals that it's done. Ultimately aren't we taking a synchronous call, calling it asynchronously, and then blocking on the main thread to join it back? Which is a fancy way of just calling it synchronously.

The relevant lines of `client.RunNode` don't do much song and dance with regard to concurrency:
```golang
	defer loggedStop(app)
	err = logConfigVariables(store) // print relevant info!
	if err != nil {
		return err
	}

	return cli.errorOut(cli.Runner.Run(app))
}
```

I am so frustrated by this mandelbug though, that I'm willing to accept it and see if it fixes it. What convinces you that this is a fix? It survived a hundred runs?